cognite/client/data_classes/_base.py

```diff
@@ -393,7 +401,9 @@ def dump(self, camel_case: bool = True) -> dict[str, Any]:
         Returns:
             dict[str, Any]: A dictionary representation of the instance.
         """
-        assert camel_case is True, "snake_case is currently unsupported"  # TODO maybe (when unifying classes)
+        if camel_case is False:
+            # TODO maybe (when unifying classes)
+            raise ValueError("snake_case is currently unsupported")
```

Comment:
Why even have the option fi it is not supported? This is not a `CogniteResource` so suggest just removing the `camel_case` parameter.

Reply:
Yeah, that's better

Reply:
Unable to remove it, as several methods call dump with `camel_case=True` on list of objects that's `Union[CogniteUpdate, CogniteResource]`. Changed the type hint to `Literal[True]` instead and removed the if-test.